.github/workflows/reusable_build.yml

```diff
@@ -578,3 +581,45 @@ jobs:
           name: ${{ inputs.target }}-${{ inputs.subtarget }}-external-toolchain
           path: openwrt/bin/targets/${{ inputs.target }}/${{ inputs.subtarget }}/${{ steps.get-toolchain-name.outputs.toolchain-name }}
           retention-days: 1
+
+  check:
+    name: Check packages for ${{ inputs.target }}/${{ inputs.subtarget }}
+    needs: build
```

Comment:
> and not after the build

I agree with that as well. The build itself is the most resource expensive process, so I would run it after this check, so it should be the opposite, build needs "Check packages".